### PR TITLE
migration_guide: add entries for SLM

### PR DESCRIPTION
**Pull request by tomi-font**
Also, make small fixes to the release notes.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst [hunk 1]
```diff
@@ -15,6 +15,16 @@ The following changes are mandatory to make your application work in the same wa
 * All references to GNSS assistance have been changed from ``A-GPS`` to `A-GNSS`_.
   This affects the :ref:`nrfxlib:nrf_modem_gnss_api`, as well as :ref:`lib_nrf_cloud`, :ref:`lib_location`, and :ref:`lib_lwm2m_location_assistance` libraries.
   See release notes for changes related to each component.
+* For the Serial LTE Modem (SLM) application:
+  * The :kconfig:option:`CONFIG_SLM_AGPS` Kconfig option has been replaced by :kconfig:option:`CONFIG_NRF_CLOUD_AGNSS` and is now enabled by default.
+  * The :kconfig:option:`CONFIG_SLM_PGPS` Kconfig option has been replaced by :kconfig:option:`CONFIG_NRF_CLOUD_PGPS`.
+  * The :kconfig:option:`CONFIG_SLM_LOCATION` Kconfig option has been replaced by :kconfig:option:`CONFIG_NRF_CLOUD_LOCATION` and is now enabled by default.
+  * The ``#XCELLPOS`` and ``#XWIFIPOS`` AT commands have been replaced by ``#XNRFCLOUDPOS``.
+  * The ``#XAGPS`` and ``#XPGPS``AT commands have been replaced by ``#XGPS``.
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst:23]:
```suggestion
  * The ``#XAGPS`` and ``#XPGPS`` AT commands have been replaced by ``#XGPS``.
```

**Review by wiba-nordic — APPROVED** [review]

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst [hunk 2]
```diff
@@ -30,6 +40,9 @@ The following changes are recommended for your application to work optimally aft
 * Applications that use Zephyr's LwM2M stack and the :ref:`lib_lwm2m_client_utils` library must refactor to use the new event :c:member:`LWM2M_FOTA_UPDATE_MODEM_RECONNECT_REQ` when updating the modem firmware to avoid rebooting the device.
   For an example, see the :ref:`lwm2m_client` sample.
 * Applications that use Zephyr's LwM2M stack are recommended to use the :kconfig:option:`CONFIG_LWM2M_UPDATE_PERIOD` Kconfig option to set the LwM2M update sending interval.
+* For the Serial LTE Modem (SLM) application:
+  * If you are using the LwM2M carrier library, make sure to take into account the addition of the auto-connect feature that is enabled by default.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst:44]:
```suggestion
  * If you are using the :ref:`liblwm2m_carrier_readme` library, make sure to take into account the addition of the auto-connect feature that is enabled by default.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]
Looks good. My changes are coming today or tomorrow.

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst [hunk 3]
```diff
@@ -15,6 +15,16 @@ The following changes are mandatory to make your application work in the same wa
 * All references to GNSS assistance have been changed from ``A-GPS`` to `A-GNSS`_.
   This affects the :ref:`nrfxlib:nrf_modem_gnss_api`, as well as :ref:`lib_nrf_cloud`, :ref:`lib_location`, and :ref:`lib_lwm2m_location_assistance` libraries.
   See release notes for changes related to each component.
+* For the Serial LTE Modem (SLM) application:
+  * The :kconfig:option:`CONFIG_SLM_AGPS` Kconfig option has been replaced by :kconfig:option:`CONFIG_NRF_CLOUD_AGNSS` and is now enabled by default.
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst:19]:
```suggestion
* For the Serial LTE Modem (SLM) application:

  * The :kconfig:option:`CONFIG_SLM_AGPS` Kconfig option has been replaced by :kconfig:option:`CONFIG_NRF_CLOUD_AGNSS` and is now enabled by default.
```
Breaks doc build by trying to apply term definition formatting.

**Pull request merged**